akka-http/src/main/scala/akka/http/scaladsl/server/directives/FutureDirectives.scala

```diff
@@ -32,6 +38,25 @@ trait FutureDirectives {
     }
 
   /**
+    * "Unwraps" a `Future[T]` and runs the inner route after future
+    * completion with the future's value as an extraction of type `T` if
+    * the supplied `CircuitBreaker` is open.
+    * If the supplied `CircuitBreaker` is open the request is rejected
+    * with a `CircuitBreakerOpenRejection`.
+    *
+   * @group future
+   */
+  def onCompleteWithBreaker[T](breaker: CircuitBreaker)(future: ⇒ Future[T]): Directive1[Try[T]] =
+    onComplete(breaker.withCircuitBreaker(future)).flatMap {
+      case Failure(ex: CircuitBreakerOpenException) ⇒
+        extractRequestContext.flatMap { ctx ⇒
+          ctx.request.entity.dataBytes.runWith(Sink.ignore)(ctx.materializer)
```

Comment:
Is it a good thing though? Since the circuit breaker is open, I find no issue with breaking the connection, too, instead of draining arbitrary number of bytes.